Client/Frontend/Home/RecentlySaved/RecentlySavedDataAdaptor.swift

```diff
@@ -7,6 +7,7 @@ import Storage
 
 protocol RecentlySavedDataAdaptor {
     func getHeroImage(forSite site: Site) -> UIImage?
+    func getFaviconImage(forSite site: Site) -> UIImage?
```

Comment:
There's tests for `getHeroImage` in RecentlySavedDataAdaptorTests, we should add some for `getFaviconImage`